pkg/api/types.go

```diff
@@ -545,6 +545,10 @@ type LiteralTestStep struct {
 	Credentials []CredentialReference `json:"credentials,omitempty"`
 	// Environment lists parameters that should be set by the test.
 	Environment []StepParameter `json:"env,omitempty"`
+	// OptionalOnSuccess defines if this step should be skipped as long
+	// as all `pre` and `test` steps were successful and AllowSkipOnSuccess
+	// flag is set to true in MultiStageTestConfiguration.
+	OptionalOnSuccess bool `json:"optional_on_success,omitempty"`
```

Comment:
Hm so technically with the addition of this here, you add it as a possible value to all `pre` and `test` steps, too, which is not part of the contract that you are describing. We should either make a separate type for `post` steps and only expose it there (my preference) or add validation that disallows it on `pre` and `test` steps.

Reply:
Hmm. Exposing it in a specific type for `post` phase would be quite some work. The LiteralTestStep is used in 101 places :-D 
To me it makes sense to have this option also for `test` steps. Would it make sense to clarify the contract and then add validation that jsut for `pre` phase where it doesn't make sense? Something like this:
```
OptionalOnSuccess defines if this step should be skipped as long as all steps from previous phases (`pre` or `test`) were successful and AllowSkipOnSuccess flag is set to true in MultiStageTestConfiguration. This option is applicable to `test` and `post` steps.
```

Reply:
I don't understand the case where it makes sense for `test` steps, can you explain? `tests` steps won't be run at all when `pre` ones fail, so having a `test` step optional on success would create a step that never runs?

Reply:
I see :-/ that's probably not what we want.

Reply:
Did this get resolved? Is it ready for another review?

Reply:
No. This will need more work.